ganga/GangaLHCb/Lib/LHCbDataset/LHCbDataset.py

```diff
@@ -110,20 +112,37 @@ def isFileTest(_file):
 
         self.persistency = persistency
         self.depth = depth
+        self.lfnList = []
         logger.debug("Dataset Created")
 
+    def setReference(self, jobNo, indices):
+        '''A function to set up the references of a subjob's dataset to the master job'''
+        self.ref = jobNo
+        self.files = indices
+
+    def createIndex(self):
+        '''Populate a list of the dataset's LFNs for some quick sorting'''
+        self.lfnList = [df.lfn for df in self.files]
```

Comment:
Will this create a very large list in memory that we then never garbage collect? Is it a problem?

Reply:
Yes. I should get the splitter to clear this after splitting.